packages/iTwinUI-react/src/core/utils/components/MiddleTextTruncation.test.tsx

```diff
@@ -59,3 +59,21 @@ it('should leave original text', () => {
   const containerSpan = container.querySelector('span') as HTMLSpanElement;
   expect(containerSpan.textContent).toBe('No trunc');
 });
+
+it('should render custom text', () => {
+  jest.spyOn(UseOverflow, 'useOverflow').mockReturnValue([jest.fn(), 20]);
+  const text = 'This is some very long text to truncate and expect ellipsis';
+  const { container } = render(
+    <MiddleTextTruncation
+      text={text}
+      textRenderer={(truncatedText) => (
+        <span>{truncatedText} - some additional text</span>
+      )}
+    />,
+  );
+
+  const containerSpan = container.querySelector('span') as HTMLSpanElement;
+  expect(containerSpan.textContent).toBe(
+    'This is some …lipsis - some additional text',
+  );
+});
```

Comment:
Maybe adding a short test to verify that the custom React node is indeed rendered. So something like how you did before, but adding the `.toBeTruthy()`

```tsx
expect(
  containerSpan.querySelector('[data-testid="custom-text"]'), // The <span> in textRenderer() has data-testid="custom-text"
).toBeTruthy();
```

and `textRenderer` on lines 69-71 to be something like

```tsx
textRenderer={(truncatedText) => (
  <span data-testid='custom-text'>{truncatedText} - some additional text</span>
)}
```

This is because the selector on line 75 is for the parent `<span>` (lines 60-70 of *MiddleTextTruncation.tsx*) and the selector written above is for the `textRenderer()`'s `<span>`